ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2

```diff
@@ -34,12 +34,29 @@ router bgp {{ router_bgp.as }}
 {%     if router_bgp.bgp.bestpath.d_path is arista.avd.defined(true) %}
    bgp bestpath d-path
 {%     endif %}
+{%     if router_bgp.listen_ranges is arista.avd.defined %}
+{%         for listen_range in router_bgp.listen_ranges | arista.avd.natural_sort('peer_group') if listen_range.peer_group is arista.avd.defined %}
```

Comment:
you have 3 levels of `if`s here. I suggest to move them into one below the `for`.
```jinja2
{% if listen_range.peer_group is arista.avd.defined and listen_range.prefix is arista.avd.defined
      and (listen_range.peer_filter is arista.avd.defined or listen.range.remote_as is arista.avd.defined) %}
```
processing wise they will still be evaluated one by one, and if one is false, the other's won't be evaluated.

Reply:
thanks! I wanted to avoid a long oneliner, but with the linebreak it looks ok!